ironfish/src/wallet/walletdb/walletdb.ts

```diff
@@ -604,6 +660,61 @@ export class WalletDB {
       [account.prefix, [assetId, [sequence, [value, noteHash]]]],
       tx,
     )
+    await this.deleteValueToUnspentNoteHash(account, decryptedNote, tx)
+  }
+
+  async *loadUnspentNoteHashesByValue(
+    account: Account,
+    assetId: Buffer,
+    reverse = false,
+    start = 0n,
+    end = 2n ** 64n - 1n,
+    tx?: IDatabaseTransaction,
+  ): AsyncGenerator<DecryptedNoteValue> {
+    for await (const noteHash of this.loadSortedUnspentNoteHashes(
+      account,
+      assetId,
+      reverse,
+      start,
+      end,
+      tx,
+    )) {
+      const decryptedNote = await this.decryptedNotes.get([account.prefix, noteHash], tx)
+
+      if (decryptedNote !== undefined) {
+        yield decryptedNote
+      }
+    }
+  }
+
+  async *loadSortedUnspentNoteHashes(
```

Comment:
This function is actually what the above function name is doing. The precedence is to use `loadIndexName` so I think this should be `loadValueToUnspentNoteHashes` since all this is is an index loading function.

Reply:
I ended up removing this function and updating the one in `ironfish/src/wallet/account/account.ts` to include the logic of checking whether the decrypted note is undefined.